File: cft-integration/src/de/proskor/fel/impl/EntityImpl.java
package de.proskor.fel.impl;

import de.proskor.fel.Entity;

abstract class EntityImpl implements Entity {
	private String name;
	private String guid;
	private int id;

	@Override
	public String getName() {
		return name;
	}

	@Override
	public int getId() {
		return id;
	}

	@Override
	public String getGuid() {
		return guid;
	}
	
	public EntityImpl(String name, String guid, int id) {
		this.name = name;
		this.guid = guid;
		this.id = id;
	}
}

File: fel-adapter/src/de/proskor/fel/impl/EntityImpl.scala
package de.proskor.fel.impl

import de.proskor.automation.Element
import de.proskor.fel.Entity;

class EntityImpl(peer: Element) extends Entity {
  override def getId: Int = peer.id
  override def getGuid: String = peer.guid
  override def getName: String = peer.name

  override def getAuthor: String = peer.author
  override def setAuthor(author: String) {
    peer.author = author
  }

  override def getDescription: String = peer.description
  override def setDescription(description: String) {
    peer.description = description
  }

  override def getQualifiedName: String = "TODO"
}